src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -23,9 +23,19 @@ public partial class Socket : IDisposable
 
         private SafeSocketHandle _handle;
 
-        // _rightEndPoint is null if the socket has not been bound.  Otherwise, it is any EndPoint of the
+        // RightEndPoint is null if the socket has not been bound.  Otherwise, it is any EndPoint of the
         // correct type (IPEndPoint, etc).
-        internal EndPoint? _rightEndPoint;
+        private EndPoint? _rightEndPoint;
+        internal EndPoint? RightEndPoint
+        {
+            get { return _rightEndPoint; }
+            private set
+            {
+                _rightEndPoint = value;
+                _localEndPoint = null;
```

Comment:
I realized this might be an unnecessary step in many cases. Which tests do fail when you remove this line?

Reply:
So actually none of them do 🤔 it seems strange and maybe we are lacking some tests? I am thinking that maybe we should clear `_localEndPoint` if new `_rightEndPoint` assigned is of different type than the old `_rightEndPoint` . However I'm not sure if this is a possible scenario? If it is, we should create a test for that.

Reply:
I couldn't find any case in which original LocalEndPoint (without cache) would start to return a different value. Removed clearing _localEndPoint.